cpp/modmesh/buffer/pymod/wrap_SimpleArray.cpp

```diff
@@ -141,24 +141,15 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
             .def_property_readonly("nbody", &wrapped_type::nbody)
             .def_property_readonly("plex", [](wrapped_type const & arr)
                                    { return pybind11::cast(SimpleArrayPlex(arr)); })
-            .wrap_modifiers()
+            .def(
+                "fill", [](wrapped_type & arr, value_type const value)
```

Comment:
it's interesting that we need to write in this way to make pybind11 work correctly.

Reply:
This is interesting, and I used some similar workarounds before with boost.python.

Does using lambda resolve the bad assembly?  If yes, please add comments here to explain to future maintainers.  Also, try to keep `wrap_modifiers()`.

Reply:
sure, kept `wrap_modifiers` and used lambda inside.

Reply:
1. The lambda isn't responsible for the problematic assembly; instead, putting `SimpleArrayCaster.hpp` into `array_common.hpp` addresses it (I don't know why this works, to be honested)
2. Upon integrating SimpleArrayCaster.hpp into wrap_SimpleArray.cpp, a lambda becomes necessary to rectify the erroneous implicit casting in Pybind11. Without the lambda, Pybind11 fails to build due to certain type issues, albeit not explored in depth.

Reply:
Oh this is even more interesting.  There is definitely something we missed or misunderstood in the language and/or the compiler.  But pragmatism beats purity.  Let's get this to work and keep the experience.  Maybe someday the Lord will shed lights on this matter.

This is good finding and description, and will be a good reference in the future.